### PR TITLE
[SPARK-29226][BUILD] Upgrade jackson-databind to 2.9.10 and fix vulnerabilities.

### PR DESCRIPTION
**Pull request by beliefer**
### What changes were proposed in this pull request?
The current code uses com.fasterxml.jackson.core:jackson-databind:jar:2.9.9.3 and it will cause a security vulnerabilities. We could get some security info from https://www.tenable.com/cve/CVE-2019-16335 and https://www.tenable.com/cve/CVE-2019-14540

This reference remind to upgrate the version of `jackson-databind` to 2.9.10 or later.

This PR also upgrade the version of jackson to 2.9.10.

### Why are the changes needed?
This PR fix the security vulnerabilities.

### Does this PR introduce any user-facing change?
No.

### How was this patch tested?
Exists UT.

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -170,8 +170,8 @@
     <!-- for now, not running scalafmt as part of default verify pipeline -->
     <scalafmt.skip>true</scalafmt.skip>
     <codehaus.jackson.version>1.9.13</codehaus.jackson.version>
-    <fasterxml.jackson.version>2.9.9</fasterxml.jackson.version>
-    <fasterxml.jackson.databind.version>2.9.9.3</fasterxml.jackson.databind.version>
+    <fasterxml.jackson.version>2.9.10</fasterxml.jackson.version>
+    <fasterxml.jackson.databind.version>2.9.10</fasterxml.jackson.databind.version>
     <snappy.version>1.1.7.3</snappy.version>
```
Comment (wangyum) [inline pom.xml]: Please update the manifest file: `./dev/test-dependencies.sh --replace-manifest`

Comment (beliefer) [reply pom.xml]: OK. Thanks!

**Review by srowen — COMMENTED** [review]
Yep looks OK pending updating the manifest, and tests.

**Comment by dongjoon-hyun** [comment]
Hi, @beliefer . I updated the PR title tag from `[CORE]` to `[BUILD]`.

**Review by dongjoon-hyun — APPROVED** [review]

**Comment by dongjoon-hyun** [comment]
Merged to master.

**Comment by beliefer** [comment]
@dongjoon-hyun @srowen @wangyum Thanks for all your help.

**Pull request closed without merging**